src/mongocxx/write_concern.cpp

```diff
@@ -65,9 +65,20 @@ void write_concern::nodes(std::int32_t confirm_from) {
 }
 
 void write_concern::acknowledge_level(write_concern::level confirm_level) {
-    libmongoc::write_concern_set_w(
-        _impl->write_concern_t,
-        static_cast<std::underlying_type<write_concern::level>::type>(confirm_level));
+    std::int32_t w;
+    switch (confirm_level) {
+    case write_concern::level::k_unacknowledged:
+        w = 0;
+        break;
+    case write_concern::level::k_majority:
+        w = -3;
+        break;
+    case write_concern::level::k_default:
+    default:
```

Comment:
Nice use of fallthrough. However, I wonder if we want setting it here to anything other than k_un..., k_majority, or k_default to actually be an exception? Or, convince me that defaulting to default is the right thing to do.